src/Val.cc

```diff
@@ -3223,9 +3223,15 @@ ValPtr TypeVal::DoClone(CloneState* state)
 	return {NewRef{}, this};
 	}
 
-VectorVal::VectorVal(VectorTypePtr t) : Val(t)
+VectorVal::VectorVal(VectorTypePtr t) :
+VectorVal(t, new vector<std::optional<ZVal>>())
 	{
-	vector_val = new vector<std::optional<ZVal>>();
+	}
+
+VectorVal::VectorVal(VectorTypePtr t, std::vector<std::optional<ZVal>>* vals)
```

Comment:
I feel like this shouldn't take a bare pointer in some way. I'd need to look at the implications of changing `vector_val` to be a `std::unique_ptr` though.

Reply:
I left this be for now, as bare pointers is how these vectors are currently being handled inside `Val.cc`.

Reply:
(also, seems okay to me as the memory management is straightforward to get right manually)